wallet-new/src/Cardano/Wallet/WalletLayer.hs

```diff
@@ -367,15 +352,71 @@ instance Buildable GetTxError where
     build (GetTxUnknownHdAccount err) =
         bprint ("GetTxUnknownHdAccount " % build) err
 
+instance Exception GetTxError
 
-instance Arbitrary GetTxError where
-    arbitrary = oneof [ pure GetTxMissingWalletIdError
-                      , pure (GetTxAddressDecodingFailed "by_amount")
-                      , pure (GetTxInvalidSortingOperation "123")
-                      , GetTxUnknownHdAccount <$> arbitrary
-                      ]
+------------------------------------------------------------
+-- Active wallet errors
+------------------------------------------------------------
 
-instance Exception GetTxError
+data NewPaymentError =
+      NewPaymentError Kernel.PaymentError
+    | NewPaymentTimeLimitReached TimeExecutionLimit
+    | NewPaymentWalletIdDecodingFailed Text
+    | NewPaymentUnknownAccountId Kernel.UnknownHdAccount
+
+-- | Unsound show instance needed for the 'Exception' instance.
+instance Show NewPaymentError where
+    show = formatToString build
+
+instance Exception NewPaymentError
+
+instance Buildable NewPaymentError where
+    build (NewPaymentError kernelErr) =
+        bprint ("NewPaymentError " % build) kernelErr
+    build (NewPaymentTimeLimitReached ter) =
+        bprint ("NewPaymentTimeLimitReached " % build) ter
+    build (NewPaymentWalletIdDecodingFailed txt) =
+        bprint ("NewPaymentWalletIdDecodingFailed " % build) txt
+    build (NewPaymentUnknownAccountId err) =
+        bprint ("NewPaymentUnknownAccountId " % build) err
+
+
+data EstimateFeesError =
+      EstimateFeesError Kernel.EstimateFeesError
+    | EstimateFeesTimeLimitReached TimeExecutionLimit
+    | EstimateFeesWalletIdDecodingFailed Text
+
+-- | Unsound show instance needed for the 'Exception' instance.
+instance Show EstimateFeesError where
+    show = formatToString build
+
+instance Exception EstimateFeesError
+
+instance Buildable EstimateFeesError where
+    build (EstimateFeesError kernelErr) =
+        bprint ("EstimateFeesError " % build) kernelErr
+    build (EstimateFeesTimeLimitReached ter) =
+        bprint ("EstimateFeesTimeLimitReached " % build) ter
+    build (EstimateFeesWalletIdDecodingFailed txt) =
+        bprint ("EstimateFeesWalletIdDecodingFailed " % build) txt
+
+data RedeemAdaError =
+    RedeemAdaError Kernel.RedeemAdaError
+  | RedeemAdaWalletIdDecodingFailed Text
+  | RedeemAdaInvalidRedemptionCode InvalidRedemptionCode
+
+instance Show RedeemAdaError where
+    show = formatToString build
+
+instance Exception RedeemAdaError
+
+instance Buildable RedeemAdaError where
+    build (RedeemAdaError err) =
+        bprint ("RedeemAdaError " % build) err
+    build (RedeemAdaWalletIdDecodingFailed txt) =
+        bprint ("RedeemAdaWalletIdDecodingFailed " % build) txt
+    build (RedeemAdaInvalidRedemptionCode txt) =
+        bprint ("RedeemAdaInvalidRedemptionCode " % build) txt
```

Comment:
5352961fc2e86a9536f44bcd0cfaed5f706565e6 this commit? It just swaps two sections. We could drop it, but I found it needed for implementing instances for the `WalletLayer` errors to avoid Template Haskell problems.

Reply:
Okay :/ .. But the template haskell was only for the SOP-Generics right? Since we dropped it, I wouldn't do the swap either here and keep the PR just about the rest.